legate_core_cpp.cmake

```diff
@@ -257,6 +233,14 @@ set_target_properties(legate_core
                       INTERFACE_POSITION_INDEPENDENT_CODE ON
                       LIBRARY_OUTPUT_DIRECTORY            lib)
 
+if(Legion_USE_CUDA)
+  set_property(TARGET legate_core PROPERTY CUDA_ARCHITECTURES ${Legion_CUDA_ARCH})
```

Comment:
It seems like we now have a potential conflict between `CMAKE_CUDA_ARCHITECTURES` and `Legion_CUDA_ARCH`.  If someone were to specify `-DCMAKE_CUDA_ARCHITECTURES=80`, e.g., it will just be silently ignored because of this line.

Or is this okay because of the new line in Legion:
```
    if(CMAKE_CUDA_ARCHITECTURES)
      set(Legion_CUDA_ARCH ${CMAKE_CUDA_ARCHITECTURES})
```
?

Reply:
Yeah right now if someone sets `-DCMAKE_CUDA_ARCHITECTURES=` (or has set it before `add_subdirectory(./legion)`), then I made Legion use it instead of `-DLegion_CUDA_ARCH=`.

It feels a bit backward at first since `-DLegion_CUDA_ARCH` seems more project-specific than `-DCMAKE_CUDA_ARCHITECTURES=`. But `Legion_CUDA_ARCH` is parsed (commas -> semis) and defaults to `all-major` if unset, so I wanted to have an "escape hatch" if someone wants to force an exact list of GPU architectures that skips (or resets) all of Legion's arch logic.

Reply:
Oh but you're right -- if `find_package(Legion)` succeeds, now legate.core will ignores the `-DCMAKE_CUDA_ARCHITECTURES` argument and use whatever archs Legion uses. This mirrors how if we find a Legion that built with `-DLegion_USE_CUDA=OFF`, legate.core uses that value even if the user configures with `-DLegion_USE_CUDA=ON`.

Reply:
Hmm... I think that's fine. I guess might at least want a warning in that situation. Or do we want to FATAL_ERROR if someone tries to set `CMAKE_CUDA_ARCHITECTURES` in that situation and it's totally disallowed?